### PR TITLE
implement varlink createfromcontainer (aka commit)

### PR DESCRIPTION
**Pull request by baude**
Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: libpod/container_commit.go [hunk 1]
```diff
@@ -145,6 +144,5 @@ func (c *Container) Commit(ctx context.Context, destImage string, options Contai
 	if err != nil {
 		return nil, err
 	}
-	fmt.Fprintf(commitOptions.ReportWriter, importBuilder.Comment())
```
Comment (TomSweeneyRedHat) [inline libpod/container_commit.go]: Is this removing the display on the container's id after the commit completes?

Comment (baude) [reply libpod/container_commit.go]: No printing in libpod

File: API.md [hunk 2]
```diff
@@ -137,12 +137,18 @@ This function is not implemented yet.
...
 
 method CreateContainer() [NotImplemented](#NotImplemented)</div>
-This method has not been implemented yet.
+This method is not implemented yet.
 ### <a name="CreateFromContainer"></a>func CreateFromContainer
 <div style="background-color: #E8E8E8; padding: 15px; margin: 10px; border-radius: 10px;">
 
-method CreateFromContainer() [NotImplemented](#NotImplemented)</div>
-This method is not implemented.
+method CreateFromContainer(name: [string](https://godoc.org/builtin#string), image_name: [string](https://godoc.org/builtin#string), changes: [[]string](#[]string), author: [string](https://godoc.org/builtin#string), message: [string](https://godoc.org/builtin#string), pause: [bool](https://godoc.org/builtin#bool)) [string](https://godoc.org/builtin#string)</div>
```
Comment (rhatdan) [inline API.md]: Why not call it commit?

File: API.md [hunk 3]
```diff
@@ -137,12 +137,18 @@ This function is not implemented yet.
...
+CreateImageFromContainer, also known as Commit, creates an image from an existing container. It requires the name or
```
Comment (rhatdan) [inline API.md]: Is this CreateFromContainer or CreateImageFromContainer?

File: API.md [hunk 4]
```diff
@@ -575,7 +581,7 @@ name [string](https://godoc.org/builtin#string)
 star_count [int](https://godoc.org/builtin#int)
 ### <a name="InfoGraphStatus"></a>type InfoGraphStatus
 
-InfoGraphStatus describes the detailed status of the graphc
+InfoGraphStatus describes the detailed status of the graph
```
Comment (rhatdan) [inline API.md]: What is a graph?

File: cmd/podman/varlink/io.projectatomic.podman.varlink [hunk 5]
```diff
@@ -391,8 +391,14 @@ method SearchImage(name: string, limit: int) -> (images: []ImageSearch)
...
-# This method is not implemented.
-method CreateFromContainer() -> (notimplemented: NotImplemented)
+# CreateImageFromContainer, also known as Commit, creates an image from an existing container. It requires the name or
+# ID of the container as well as the resulting image name.  Optionally, you can define an author and message
+# to be added to the resulting image.  You can also define changes to the resulting image for the following
+# attributes: _CMD, ENTRYPOINT, ENV, EXPOSE, LABEL, STOPSIGNAL, USER, VOLUME, and WORKDIR_.  To pause the
+# container while it is being committed, pass a _true_ bool for the pause argument.  If the container cannot
+# be found by the ID or name provided, a (ContainerNotFound)[#ContainerNotFound] error will be returned; otherwise,
+# the resulting image's ID will be returned as a string.
+method CreateFromContainer(name: string, image_name: string, changes: []string, author: string, message: string, pause: bool) -> (image: string)
```
Comment (rhatdan) [inline cmd/podman/varlink/io.projectatomic.podman.varlink]: CreateFromContainer or CreateImageFromContainer?

File: API.md (hunk 4 above)
Comment (mheon) [reply API.md]: We could call it "graph driver" to match Docker, but I'd personally prefer calling it something like StorageDriver

Comment (rhatdan) [reply API.md]: Sounds much better to me.  Graph is a very internal name.

File: API.md (hunk 2 above)
Comment (baude) [reply API.md]: done

File: API.md (hunk 3 above)
Comment (baude) [reply API.md]: its commit now

File: API.md (hunk 4 above)
Comment (baude) [reply API.md]: storage driver it is

File: cmd/podman/varlink/io.projectatomic.podman.varlink (hunk 5 above)
Comment (baude) [reply cmd/podman/varlink/io.projectatomic.podman.varlink]: it is commit

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
Actually @baude Can you change the commit message to say you are implementing commit.

**Comment by mheon** [comment]
@baude Lint errors

**Comment by mheon** [comment]
LGTM once the lint error is gone

**Comment by baude** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**